client/lib/yourturn/src/notification-bar/notification-bar.jsx

```diff
@@ -34,17 +34,38 @@ class NotificationBar extends React.Component {
     dismiss(id) {
         this.props.dispatch(Actions.removeNotification(id));
     }
+    
+    componentWillReceiveProps(nextProps) {
+        const { notifications = [] } = nextProps;
+
+        if (!this.state.interval && notifications.length > 0) {
+            let interval = setInterval(() => {
```

Comment:
This code is duplicated with what's in `componentDidMount`. Can you refactor this in two functions `startCountdown` and `stopCountdown` or something?